test/Npgsql.Tests/PrepareTests.cs

```diff
@@ -53,6 +58,43 @@ public async Task Async()
             }
         }
 
+        [Test, IssueLink("https://github.com/npgsql/npgsql/issues/2925")]
+        public async Task Bug2925()
+        {
+            if (Type.GetType("Mono.Runtime") != null)
+                Assert.Ignore("Flaky on mono");
+
+            var builder = new NpgsqlConnectionStringBuilder(ConnectionString)
+            {
+                WriteBufferSize = NpgsqlWriteBuffer.MinimumSize,
+            };
+            await using var postmasterMock = new PgPostmasterMock(builder.ConnectionString);
+            using var _ = CreateTempPool(postmasterMock.ConnectionString, out var connectionString);
+            await using var conn = await OpenConnectionAsync(connectionString);
+            var server = postmasterMock.GetPendingServer();
+
+            var sb = new StringBuilder();
+            for (var i = 0; i < 1000; i++)
+            {
+                sb.Append($"SELECT {i};");
+            }
+            using var cmd = new NpgsqlCommand(sb.ToString(), conn);
+            // Async read wouldn't complete until async write from prepare is completed (on linux, on windows is ok)
+            // So, to make them run in parallel, we put it in another thread
+            var breakThread = new Thread(server.BreakOnRead);
+            breakThread.Start();
+            try
+            {
+                Assert.That(() => cmd.Prepare(), Throws.TypeOf<NpgsqlException>()
```

Comment:
I don't think there's any guarantee that this tests failures from the *write* task, no? Couldn't this be catching exceptions from reading instead?

Reply:
Yep, you're correct. But there is nothing really we can do right now.